### ISSUE TITLE
amp-next-page - max-pages and recommendation-box

### ISSUE THREAD

**Issue by smatoox** [ampproject/amphtml#28097]
## What's the issue?

I see strange behaviors with recommandation box for amp-next-page :

### **1. List of entries and max-pages attribute**
For example i'm using a json with 4 entries and I have an attribute `max-pages="2"`.
Render displays correctly my 2 entries.
But I have not the `recommandation box` displayed for 2 last entries.

Is it normal ? Should we display rest of list in `recommandation box` automatically ?

### **2. Entries + max-pages + template custom recommandation box**
In the same code, I add custom template (copy/paste from documentation).

**2.1 Error when using template**
In local test, an error appears, but not on a server like codesandbox and only for recommandation box and not for separator template : 
> Uncaught (in promise) Error: Template not found for div​​​
![Capture d’écran de 2020-04-29 09-30-51](https://user-images.githubusercontent.com/26866646/80574627-7e321300-8a02-11ea-9fe8-ccb9d3110176.png)

**2.1 Recommandation box displays entries with custom template**
In my first point, i had not recommandation box displayed.
But with custom template we have it and content is very strange and random.

In 4 refresh, recommandation box displayed:

- (3x) the first and the second entries already displayed above the box
- (1x) only the second entry already displayed too above the box

On rendered html I can see error's class :
![Capture d’écran de 2020-04-29 10-23-17](https://user-images.githubusercontent.com/26866646/80576020-bfc3bd80-8a04-11ea-949a-2b42a52486f4.png)

Why recommandation box contains entries already displayed correctly ?

This 3 conditions are not respected with the exception of the last :

> The user reaches the end of the page before the next page had loaded.
> The next page fails to load.
> If the max-pages attribute is specified and the number of displayed pages is met.

But why we have same entries already displayed ?

## How do we reproduce the issue?

Code is available on code sand box : https://codesandbox.io/embed/strange-water-dqtdr?fontsize=14&hidenavigation=1&theme=dark

## What browsers are affected?

Tested on Linux 18.04.4 - Chrome 81.0.4044.122

**Comment by wassgha** [comment]
I belive most of these issues have been fixed by https://github.com/ampproject/amphtml/pull/27764 which has not reached the production build but can be tested in canary. Could you please try it and update the issue with any remaining bugs?

**Comment by smatoox** [comment]
Indeed all errors and strange behaviors are gone.
Thanks @wassgha

**Comment by wassgha** [comment]
Glad to hear that!
